### ISSUE TITLE
[Bug]: BAxUS increase_embedding_and_observations bugs

### ISSUE THREAD

**Issue by Thomas-Christie** [pytorch/botorch#3200]
Hi,

Thanks for the library! I was looking at the [BAxUS tutorial](https://botorch.org/docs/tutorials/baxus/) associated with [Increasing the Scope as You Learn:
Adaptive Bayesian Optimization in Nested Subspaces](https://openreview.net/pdf?id=e4Wf6112DI) and I believe there are two bugs in the `increase_embedding_and_observations` function. 

### Bug 1

According to the definition in the paper, a matrix $S$ is said to be a *sparse embedding matrix* if an only if each column in $S$ has exactly one non-zero entry. However, from the current run in the tutorial, I see the following:

```
S before increase
tensor([[-1.,  0.,  0.,  0.,  0.,  0., -1., -1., -1., -1.],
      [ 0.,  1.,  1.,  1.,  1.,  1.,  0.,  0.,  0.,  0.]],
     dtype=torch.float64)
X before increase
tensor([[51, 95],
      [66, 77],
      [23, 20],
      [48, 38],
      [ 0, 47],
      [22, 36],
      [33, 45]])
S after increase
tensor([[ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0., -1., -1.],
      [ 0.,  0.,  0.,  1.,  1.,  0.,  0.,  0.,  0.,  0.],
      [ 0.,  0.,  0.,  0.,  0.,  0., -1., -1.,  0.,  0.],
      [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.],
      [ 0.,  1.,  0.,  0.,  0.,  1.,  0.,  0.,  0.,  0.],
      [ 0.,  0.,  1.,  0.,  0.,  0.,  0.,  0.,  0.,  0.]],
     dtype=torch.float64)
X after increase
tensor([[51, 95, 51, 51, 95, 95],
      [66, 77, 66, 66, 77, 77],
      [23, 20, 23, 23, 20, 20],
      [48, 38, 48, 48, 38, 38],
      [ 0, 47,  0,  0, 47, 47],
      [22, 36, 22, 22, 36, 36],
      [33, 45, 33, 33, 45, 45]])
```
with the first column consisting of all zeros.

This bug occurs due to the padding column trick. When `pad_sequence` pads a shorter sequence, it fills with 0. Then `new_bins_padded + 1` shifts everything by 1, so padding zeros become 1. But a real index of 0 also becomes 1 after the shift.

Corresponding to the example above, row 0's indices $[0, 6, 7, 8, 9]$ might get split into 3 bins:
- $[8, 9]$ (stay in original row)
- $[7, 6]$
- $[0]$

Then:
```
new_bins_padded = pad_sequence([[7, 6], [0]], batch_first=True)
# = [[7, 6],
#    [0, 0]]   ← real 0, then padding 0
```
and
```
new_bins_padded + 1
# = [[8, 7],
#    [1, 1]]   ← collision - both point to column 1
```
then
```
els_to_move_padded
# = [[-1, -1],
#    [-1,  0]]  ← real -1, then padding 0
```

Then the `scatter_` writes -1 to column 1, then the padding overwrites it with 0. After `[:, 1:]`, column 0 of the result is zero, and dimension 0 is gone. I'd be happy to submit a PR with a fix for this if you want? The main change is to apply the `+1` operation which is currently in `S_stack = S_stack.scatter_(1, new_bins_padded + 1, els_to_move_padded)` before splitting/padding, so you'd get:

```
new_bins = torch.tensor_split(idxs_non_zero + 1, n_row_bins)[1:] <-- Add 1 here instead
...
S_stack = S_stack.scatter_(
     1, new_bins_padded, els_to_move_padded.to(S_stack.dtype)   <-- No longer add 1 here
)
...
S_update[row_idx, torch.hstack(new_bins) - 1] = 0 <-- Now need to subtract 1 here
```

### Bug 2

Unless I'm mistaken, I thought that `n_new_bins` in the paper was the (maximum) number of new bins *added to* each existing bin (i.e. for each row in the original $S$, we want to add (a maximum of) `n_new_bins` new rows *in addition to* this row? In this case shouldn't the line:
```
n_row_bins = min(
   n_new_bins, len(idxs_non_zero)
)
```
in fact be
```
n_row_bins = min(
   n_new_bins+1, len(idxs_non_zero)
)
```
?

cc @lpapenme 

Thanks in advance! I'm happy to fix a PR to fix both these bugs if you agree that they are both bugs.

### BoTorch Version

Latest

### Code of Conduct

- [x] I agree to follow BoTorch's Code of Conduct

**Comment by saitcakmak** [comment]
Hi @Thomas-Christie. Thanks for reporting! These do look like actual bugs to me and we'd welcome a PR fixing these

**Comment by Thomas-Christie** [comment]
Have opened https://github.com/meta-pytorch/botorch/pull/3204 to address this @saitcakmak
